internal/safrole/state.go

```diff
@@ -42,6 +44,39 @@ func (state State) RingVerifier() (*bandersnatch.RingVrfVerifier, error) {
 	return ringVerifier, nil
 }
 
+// Takes a private bandersnatch key and returns a new RingVrfProver for this state.
+func (state State) RingProver(privateKey crypto.BandersnatchPrivateKey) (*bandersnatch.RingVrfProver, error) {
+	publicKey, err := bandersnatch.Public(privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	ring := make([]crypto.BandersnatchPublicKey, len(state.NextValidators))
+	for i, vd := range state.NextValidators {
+		ring[i] = vd.Bandersnatch
+	}
+
+	// Find the prover index in the ring, if we don't find a match we return an
+	// error.
+	proverIdx := 0
+	found := false
+	for i, pk := range ring {
+		if pk == publicKey {
+			proverIdx = i
+			found = true
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("private key is not a ring member")
+	}
+
+	ringProver, err := bandersnatch.NewRingProver(privateKey, ring, uint(proverIdx))
```

Comment:
This can be written like this for short, but if you check for error you could also wrap the error
```suggestion
	return bandersnatch.NewRingProver(privateKey, ring, uint(proverIdx))
```